uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Test/Fabricator/SicknessRequest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_BAO_SicknessRequest as SicknessRequest;
+use CRM_HRLeaveAndAbsences_Test_Fabricator_LeaveBalanceChange as LeaveBalanceChangeFabricator;
+
+class CRM_HRLeaveAndAbsences_Test_Fabricator_SicknessRequest {
+
+  private static $leaveRequestStatuses;
+  private static $dayTypes;
+  private static $sicknessReasons;
+
+  public static function fabricate($params, $withBalanceChanges = false) {
```

Comment:
Because I would be using $params for something different from its original purpose. That parameter should only contain data to build the Entity and nothing else.

Also, using a separate param, I can easily declare a default value. If I as to use $params instead of $withBalanceChange, I would need to add it to a "defaultParams" array and then merge it into the $params array and, as you mentioned, remember to manually unset it.